sdk-request-identity/src/main/java/dev/restate/sdk/auth/signing/RestateRequestIdentityVerifier.java

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
+//
+// This file is part of the Restate Java SDK,
+// which is released under the MIT license.
+//
+// You can find a copy of the license in file LICENSE in the root
+// directory of this repository or package, or at
+// https://github.com/restatedev/sdk-java/blob/main/LICENSE
+package dev.restate.sdk.auth.signing;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.Ed25519Verifier;
+import com.nimbusds.jose.jwk.Curve;
+import com.nimbusds.jose.jwk.OctetKeyPair;
+import com.nimbusds.jose.util.Base64URL;
+import com.nimbusds.jwt.SignedJWT;
+import dev.restate.sdk.auth.RequestIdentityVerifier;
+import java.util.Base64;
+
+public class RestateRequestIdentityVerifier implements RequestIdentityVerifier {
+  private static final String SIGNATURE_SCHEME_HEADER = "x-restate-signature-scheme";
+  private static final String SIGNATURE_SCHEME_V1 = "v1";
+  private static final String SIGNATURE_SCHEME_UNSIGNED = "unsigned";
+  private static final String JWT_HEADER = "x-restate-jwt-v1";
+  private static final String IDENTITY_V1_PREFIX = "publickeyv1_";
+  private static final byte[] ASN1_PREFIX = Base64.getDecoder().decode("MCowBQYDK2VwAyEA");
+
+  private final JWSVerifier verifier;
+
+  private RestateRequestIdentityVerifier(JWSVerifier verifier) {
+    this.verifier = verifier;
+  }
+
+  @Override
+  public void verifyRequest(Headers headers) throws Exception {
+    String signatureScheme = expectHeader(headers, SIGNATURE_SCHEME_HEADER);
+    switch (signatureScheme) {
+      case SIGNATURE_SCHEME_V1:
+        String jwtHeader = expectHeader(headers, JWT_HEADER);
+        SignedJWT signedJWT = SignedJWT.parse(jwtHeader);
+        if (!signedJWT.verify(verifier)) {
+          throw new IllegalStateException("Verification of JWT token failed");
+        }
+        break;
+      case SIGNATURE_SCHEME_UNSIGNED:
+        break;
```

Comment:
won't this mean that I can just provide a unsigned scheme and sneak requests through despite verification? In typescript we fail here

Reply:
My bad, let me fix it